src/crimson/common/tri_mutex.cc

```diff
@@ -86,6 +87,8 @@ void tri_mutex::demote_to_read()
   assert(exclusively_used);
   exclusively_used = false;
   ++readers;
+  assert(!writers && !exclusively_used);
```

Comment:
If you move this up one line, it could actually assert !readers as well, I believe.

Reply:
Done:-)